src/Controls/tests/BindingSourceGen.UnitTests/BindingRepresentationGenTests.cs

```diff
@@ -167,6 +167,36 @@ public void GenerateBindingWithNullableSourceReferenceAndNullableReferenceElemen
         Assert.Equivalent(expectedBinding, actualBinding, strict: true);
     }
 
+    [Fact]
+    public void GenerateBindingWithNullablePropertyReferenceWhenNullableEnabled()
+    {
+        var source = """
+        using Microsoft.Maui.Controls;
+        var label = new Label();
+        label.SetBinding(Label.RotationProperty, static (Foo f) => f.Value);
+
+        class Foo
+        {
+            public string? Value { get; set; }
+        }
+        """;
+
+        var actualBinding = SourceGenHelpers.GetBinding(source);
+        var expectedBinding = new CodeWriterBinding(
+                new SourceCodeLocation("", 3, 7),
+                new TypeDescription("global::Foo"),
+                new TypeDescription("string", IsNullable: true),
+                [
+                    new ConditionalAccess(new MemberAccess("Value")),
+                ],
+                GenerateSetter: true
+            );
+
+        //TODO: Change arrays to custom collections implementing IEquatable
```

Comment:
Do we need to address this TODO? It seems to be working just fine.

Reply:
This TODO is more about perf than tests. It will make tests better as well, but major concern is caching.

Reply:
OK